testonly/hammer/hammer.go

```diff
@@ -284,6 +285,75 @@ func newWorker(cfg *MapConfig, prng *rand.Rand) *mapWorker {
 	}
 }
 
+// readWorker performs read-only operations on a fixed map.
+type readWorker struct {
+	*mapWorker
+
```

Comment:
optional nit: I'd not bother with empty lines here and in the constructor. Passim for `writeWorker`.

Reply:
I'm exercising the "do nothing" branch of "optional" here - it makes it clearer to me what is going on when the the linebreak is kept. My attitude will likely change as the Go embedding pattern instead of a more classical inheritance pattern becomes normal to me.